File: src/main/java/ir/reservs/reservs/ui/main/settings/SettingsFragment.java
package ir.reservs.reservs.ui.main.settings;

import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.cardview.widget.CardView;
import androidx.navigation.fragment.NavHostFragment;

import javax.inject.Inject;

import ir.reservs.reservs.R;
import ir.reservs.reservs.ui.base.BaseFragment;
import ir.reservs.reservs.ui.login.LoginActivity;

public class SettingsFragment extends BaseFragment implements SettingsContract.View {
    @Inject
    SettingsPresenter settingsPresenter;

    private TextView txtName;
    private TextView txtPhone;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater,
                             @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.layout_settings, container, false);
    }

    @Override
    public void onDestroyView() {
        if (settingsPresenter != null) {
            settingsPresenter.onDetach();
            settingsPresenter = null;
            txtName = null;
            txtPhone = null;
        }
        super.onDestroyView();
    }

    @Override
    public void setUserInfo(String name, String phone, String image) {
        txtName.setText(name);
        txtPhone.setText(phone);
        //todo:load image into imgAvatar
    }

    @Override
    public void openLoginActivity() {
        startActivity(new Intent(getActivity(), LoginActivity.class));
        getActivity().finish();
    }

    @Override
    public void onError(String msg) {

    }

    @Override
    public void setup(View view) {
        getFragmentComponent().inject(this);
        settingsPresenter.onAttach(this);
        txtName = view.findViewById(R.id.txtName);
        txtPhone = view.findViewById(R.id.txtPhone);
        settingsPresenter.setUserProfileData();
        //ImageView imgAvatar = view.findViewById(R.id.imgAvatar);
        CardView cardLogout = view.findViewById(R.id.cardLogout);
        CardView cardEdit = view.findViewById(R.id.cardEdit);
        CardView passwordCard = view.findViewById(R.id.passwordCard);
        cardLogout.setOnClickListener((v) -> settingsPresenter.logoutUser());
        cardEdit.setOnClickListener((v) -> NavHostFragment.findNavController(this).navigate(R.id.settingsToInformation));
        passwordCard.setOnClickListener((v) -> NavHostFragment.findNavController(this).navigate(R.id.settingsToPassword));
    }


}

File: src/main/java/ir/reservs/reservs/ui/main/settings/SettingsFragment.kt
package ir.reservs.reservs.ui.main.settings;

import android.app.AlertDialog
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.content.ContentProviderCompat.requireContext
import androidx.navigation.fragment.findNavController
import com.squareup.picasso.Picasso
import ir.reservs.reservs.R
import ir.reservs.reservs.ui.base.BaseFragment
import kotlinx.android.synthetic.main.layout_settings_new.*
import kotlinx.android.synthetic.main.layout_settings_new.imgAvatar
import kotlinx.android.synthetic.main.layout_settings_new.txtName
import kotlinx.android.synthetic.main.layout_settings_new.txtPhone
import javax.inject.Inject


class SettingsFragment : BaseFragment(), SettingsContract.View {

    var settingsPresenter: SettingsPresenter? = null
        @Inject set

    var dialog: AlertDialog? = null
        @Inject set

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.layout_settings_new, container, false)
    }

    override fun setUserInfo(name: String, phone: String, credit: Long) {
        txtName.text = name
        txtPhone.text = phone
        txtCredit.text = credit.toString()
    }

    override fun setUserImage(image: String) {
        Picasso.get().load(image).resize(128, 128).into(imgAvatar)
    }

    override fun openLoginPage() {
        //todo logout
        findNavController().backStack.clear()
        findNavController().navigate(R.id.sendFragment)
    }

    override fun setup(view: View) {
        fragmentComponent?.inject(this)
        settingsPresenter?.onAttach(this)
        settingsPresenter?.setUserProfileData()
        boxContact.setOnClickListener { findNavController().navigate(R.id.settingsToAbout) }
        boxExit.setOnClickListener { settingsPresenter?.logoutUser() }
        boxEdit.setOnClickListener { findNavController().navigate(R.id.settingsToInformation) }
        imgAvatar.setOnClickListener { settingsPresenter?.pickImage(requireActivity().supportFragmentManager, requireContext()) }
    }

    override fun showProgress() {
        dialog?.show()
    }

    override fun hideProgress() {
        dialog?.hide()
    }

    override fun onDestroyView() {
        if (settingsPresenter != null) {
            settingsPresenter?.onDetach()
            settingsPresenter = null
        }
        if (dialog != null) {
            dialog?.dismiss()
            dialog = null
        }
        super.onDestroyView()
    }
}
